akka-typed-tests/src/test/java/akka/typed/javadsl/BehaviorChainTest.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.javadsl;
+
+import akka.typed.Behavior;
+import akka.typed.Terminated;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import java.util.ArrayList;
+
+import static akka.typed.javadsl.Actor.same;
+import static akka.typed.javadsl.Actor.stopped;
+
+public class BehaviorChainTest extends JUnitSuite {
+    interface Message {
+    }
+
+    static final class One implements Message {
+        public String foo() {
+          return "Bar";
+        }
+    }
+    static final class MyList<T> extends ArrayList<T> implements Message {
+    };
+
+    @Test
+    public void shouldCompile() {
+      Behavior<Message> b = Actor.<Message>immutable()
```

Comment:
In streams we use a class parameter as an alternative to just a type parameter, which would be: `immutable(Protocol.class).match(etc)`

I'm curious why Patriks function-with-parameter-with-type-parameter would be inferred by the Java compiler if the one in this PR isn't.